### PR TITLE
Use `usize` rather than `Option<usize>` to represent `Limit::skip`and `Limit::offset`

### LINKED ISSUE #3369: Use `usize` to represent `Limit::skip`
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
```rust
pub struct Limit {
    /// Number of rows to skip before fetch
    pub skip: Option<usize>,
    /// Maximum number of rows to fetch
    pub fetch: Option<usize>,
    /// The logical plan
    pub input: Arc<LogicalPlan>,
}
```
Since `skip = 0` and `skip = None` have same meaning, we can use pure `usize` to represent `skip` to reduce complexity.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Find this will work on #3355

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

# Which issue does this PR close?

Closes #3369.

 # Rationale for this change

`skip = None` and `skip = Some(0)` are equivalent, so we don't need both of them and the `Optional` type here is redundant. We use `usize` as the type of `skip` so that `skip = None` and `skip = Some(0)` can be simplified to `skip = 0`
# What changes are included in this PR?

# Are there any user-facing changes?
Yes.

### CODE REVIEW

**Comment by HaoYang670** [comment]
Mark this as draft as I only update the logical plan so far.

**Review by andygrove — COMMENTED** [review]
This is looking good so far. Nice simplification.

File: datafusion/expr/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -1294,8 +1294,9 @@ pub struct Extension {
 #[derive(Clone)]
 pub struct Limit {
     /// Number of rows to skip before fetch
-    pub skip: Option<usize>,
-    /// Maximum number of rows to fetch
+    pub skip: usize,
+    /// Maximum number of rows to fetch,
+    /// None means fetching all rows
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:1299]: 👍

**Review by alamb — APPROVED** [review]
Looks good to me @HaoYang670  -- thanks!

fyi @turbo1912 and @AssHero and @
ming535 who have [contributed to this code](https://github.com/apache/arrow-datafusion/blame/dbfb0e6ff9bc6fb933ed7cfd15bd9127ac0d443a/datafusion/optimizer/src/eliminate_limit.rs#L57) recently

**Review by andygrove — APPROVED** [review]

File: datafusion/optimizer/src/eliminate_limit.rs [hunk 2]
```diff
@@ -82,10 +82,11 @@ fn eliminate_limit(
...
                 }
                 None => {
-                    match skip {
+                    if *skip == 0 {
                         // If there is no LIMIT and OFFSET is zero, LIMIT/OFFSET can be removed
-                        Some(skip) if *skip == 0 => return Ok(input.as_ref().clone()),
-                        _ => {}
+                        return Ok(input.as_ref().clone());
+                    } else {
+                        {}
```
Comment (jackwener) [inline datafusion/optimizer/src/eliminate_limit.rs:89]: redundant part

**Review by jackwener — APPROVED** [review]
Great job.

**Comment by jackwener** [comment]
😂Look like my review is late, new a PR for it

**Pull request merged**